### DISCUSSION TITLE
Suggestion for interface improvement

### DISCUSSION THREAD

**Discussion by IticeMotion** [friction2d/friction#39] [Ideas]
Hello, I'm a Motion Designer and would like to make some suggestions that could greatly improve the interface for better usability.  It would be great to have an Attributes/Properties window, to facilitate and speed up access without having to keep opening tabs, which is dated, another incredible improvement would be a color wheel like inkscape, to facilitate the application of colors and become more intuitive, another improvement would be a shortcut command for effects like the Fx console that with ctrl+space opens this shortcut tab.

**Comment by rodlie** [comment]
Hi

> It would be great to have an Attributes/Properties window, to facilitate and speed up access without having to keep opening tabs

What do you mean? a separate window for properties?

There are a total of 5 tabs, the default and most used are properties and fill. Do you need to flip through the tabs that often? Please explain how you want/expect the workflow to work.

![Screenshot from 2023-08-02 14-12-24](https://github.com/friction2d/friction/assets/34516798/8add6fac-d832-4004-9d83-5fa7bd1b50cc)

> another incredible improvement would be a color wheel like inkscape

Maybe ... I prefer the current color widget, a color wheel wastes space (imho). But if this is a popular request I can implement something.

> another improvement would be a shortcut command for effects like the Fx console that with ctrl+space opens this shortcut tab.

This is probably something I can implement at some point.

**Reply by rodlie** [reply]
> Yes, a color wheel'd be useful,

Maybe, not a priority for me as it require a lot of work for no gain (from my POV) and wastes a lot of space.

Feel free to make a PR (note that color widgets are OpenGL).

> specially with the default being RGB instead of HSV

RGB is default for a reason, but Friction remembers the state, so if you use HSV it will be the default.

**Comment by IticeMotion** [comment]
Hello, thanks for answering me, I think I didn't explain my point about a
properties window, when I click on the object in Friction there are still
tabs to open, it would be much more interesting if only the attributes of
the objects appear. I'll leave some examples of what I meant. Thank you for
your attention, I hope I can help more.
[image: image.png]

It could appear like these examples below without tabs.

[image: image.png]
[image: image.png]

**Reply by rodlie** [reply]
The images don't show up. You need to use the GitHub interface to add image (just drag and drop).

**Comment by dejayc** [comment]
I've only used Friction for about 10 minutes, but am immensely appreciative that it exists.  I have decades of experience using Inkscape and other software for graphics and video editing, and I believe that Friction can become an incredibly-polished application if its UI/UX conventions are tightened up.

The only real competitors to Friction, in my opinion, are paid websites like SVGator.com and Rive.app that charge $20 to $40 per month for basic features.  So making Friction as enjoyable to use as possible has the real potential to disrupt the playing field.

Just to validate my assumption that small improvements to UI/UX are impactful, please check this YouTube search for [how to save hours of effort when editing videos in DaVinci Resolve](https://www.youtube.com/results?search_query=save+hours+in+davinci+resolve).  These videos are filled with tips and tricks that individually only save a few seconds here and there, but accumulate to hours when used on a consistent basis.  The impact of small but meaningful improvements to UI/UX is real.

I would like to propose centralizing (somewhere) some discussion around UI/UX improvements to Friction, so that we can keep this dialogue open.  As an example, I can already envision a version of Friction that has the professional UI/UX of other software industry leaders, and I don't believe it would take huge amounts of effort to get there.

Does anyone have suggestions about appropriate places to have these conversations?

**Reply by rodlie** [reply]
Feel free to share your feedback in this discussion, or make a new one (can also be an issue, but they are usually reserved for specific tasks, not general discussion like here).

GitHub is currently the only place to discuss stuff around Friction (that will get my attention).

**Comment by IticeMotion** [comment]
@dejayc How about Discord or Telegram. Although Discord is more interesting for organizing rules and topics.

**Reply by dejayc** [reply]
Cool, maybe share your opinion in the [discussion about Discord](https://github.com/orgs/friction2d/discussions/111).

**Comment by rodlie** [comment]
Btw, 0.9.6 will feature some UI layout changes, see screenshots: https://github.com/friction2d/friction2d.github.io/blob/main/_drafts/friction-096.md

Will add documentation etc when I get the time.

**Reply by dejayc** [reply]
Uhm, these graphical changes look amazing, frankly.  I wouldn't be able to evaluate how effective this new UI/UX is without playing around with the UI personally, but Friction definitely seems to be heading in the right direction.

I'd like to share my hopes and concerns about the UI/UX direction of Friction.

Industry-leading incumbent software often establishes a tight grip on the industry by really getting some of the core mechanics for UI/UX right.  Challengers to the incumbents, like Inkscape, Krita, Blender, GIMP, OBS Studio, digiKam, Affinity, and DaVinci Resolve often neglect to replicate the solid, intuitive UI/UX of the incumbents.  These software applications often choose quirky, counter-intuitive UI/UX mechanisms to fill some philosophical principle that may or may not be important to end users, which usually are also irksome to industry professionals who have acclimated to decades of familiarity with industry standards.

In any event, I think that for Friction to be majorly impactful over the long term, two things need to happen:

1. Friction must understand, and adopt when possible, the premium features and UI/UX mechanics that incumbent software has already normalized
2. Friction must feel unified, cohesive, and consistent in how it implements UI/UX

I feel that plenty of feedback from users of other graphics software can contribute heavily to feature requests to Friction, ensuring that Friction gains and retains competitive functionality compared to the premium features of industry-leading software.

I also feel that Friction could benefit from having design philosophies, guiding principles, and a style guide for how UI/UX should work in Friction, to ensure that the app is self-consistent and easy to understand.  I can't tell you how many times I've almost given up on Blender, DaVince, Krita, and other software just because of lack of consistency in how UI/UX mechanics work across each app.  Sometimes, it literally takes decades for these apps to get their UI/UX right.

To further both of these goals, I also feel that Friction could benefit from having a lot of people provide feedback, constantly and with minimal friction, in a manner that facilitates discovery and brainstorming without being roadblocked by clunky communication methods such as relying upon GitHub discussion pages.  And perhaps this could be accomplished through unofficial means, such as a Discord server for Friction users.

In any event, it's been a big fantasy of mine during the past few years to eventually discover and use some graphic design solution that encompasses the best of Inkscape, OBS Studio, and DaVinci Resolve in one flexible, easy-to-use app where motions graphics "just makes sense".  Friction is the first app I've seen in awhile that gives me hope that someone will eventually get this goal right, which is why I feel passionate about providing feedback on these topics.

UPDATE: I also think a documented roadmap for Friction could help users contribute to the direction of the app.  Perhaps a solution like featureupvote.com could be used to help collect feedback, as I've seen that site used successfully for other apps and games.

**Reply by rodlie** [reply]
Thanks for the feedback, a bit too much to take in (or reply to) at this point :smile: 

The current state of UI/UX is based on usage, as it as been used in production since the start (the only reason Friction exists). Tweaks/Changes come from that usage, if something is annoying or could improve workflow (as in time is money) we try adjust to that, but it's also a matter of "cost" (as in my spare time), is the new addition worth the development time?

The target audience is not beginners, it's a tool for work, so it's assumed that the user is already a graphics designer and/or animator, and have prior experience working with similar applications.

**Comment by Dinesh0N** [comment]
G, R, S shortcuts nice 👍🏼

**Comment by Dinesh0N** [comment]
new Accent Color Options ( Ubuntu, audacity v4 )
![audacity_pkusLZNj9h](https://github.com/user-attachments/assets/06ea4937-70ff-4ac2-882e-52c920a1be6c)

**Comment by Dinesh0N** [comment]
Scroller Bar UI New functionality for ***Friction***

This attachment very useful for Tracks and Timelines
View:
![Scroller Bar UI_g24](https://github.com/user-attachments/assets/3b601739-d441-4f2b-9a86-fb0943dc8eb8)

Variation:
![Scroller Bar UI_g23](https://github.com/user-attachments/assets/7828ccee-1ff0-49ad-84b8-358f5b390a76)

Demo Video blender example:
https://github.com/user-attachments/assets/69a32c22-2167-4dbb-b790-b58c57b53028

**Reply by LPS-github** [reply]
yes please, I'd love to see that, it would be so useful!
